app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -81,34 +87,65 @@ const isElementArray = ({ max, sliceName }: IElementDefinition): boolean =>
 
 const createElementNode = (
   elementDefinition: IElementDefinition,
-  isArrayItem?: boolean
+  params: { index?: number; parentPath?: string }
 ): ElementNode => {
+  const { index, parentPath } = params;
+  const prefixPath =
+    (parentPath || elementDefinition.path?.split(".").shift()) ?? "";
+  const suffixPath = elementDefinition.path?.split(".").slice(1).join(".");
+  const elementPath = `${prefixPath}${
+    prefixPath && suffixPath && "."
+  }${suffixPath}${index !== undefined ? `[${index}]` : ""}`;
   return {
     id: uuid(),
     name: elementDefinition.id?.split(".").pop() ?? "",
     children: [],
-    path: elementDefinition.path ?? "",
+    path: elementPath,
     definition: elementDefinition,
-    isArray:
-      undefined !== isArrayItem
-        ? !isArrayItem
-        : isElementArray(elementDefinition),
+    isArray: index !== undefined ? false : isElementArray(elementDefinition),
     isSlice: !!elementDefinition.sliceName,
     kind: getKind(elementDefinition),
     type: elementDefinition.type?.map((t) => computeType(t)).join(" | "),
   };
 };
 
-const getChildrenChoices = (
+const getNewPath = (path: string): string => {
+  if (path.endsWith(`]`)) {
+    const splittedPath = path.split(".");
+    const lastPathElement = splittedPath.pop()?.split("[")[0];
+    if (lastPathElement) splittedPath.push(lastPathElement);
+    return splittedPath.join(".");
+  } else {
+    return path;
+  }
+};
+
+const createElementDefinition = (attribute: Attribute): IElementDefinition => {
+  const elementDefinition: IElementDefinition = {
+    path: attribute.path,
+    id: getNewPath(attribute.path),
```

Comment:
Why not just setting the `attribute.path` in `id` attribute?

Reply:
If you have an attribute which is an array item, the path will be ending with a "[0]" or a "[n]" where n is going to be the its index (like Observation.identifier[0]), you need to remove the [n] from the path to put it in the element definition.

Reply:
I just realised that it's only working for the last element of the path, meaning if you have a Observation.identifier[0].type.coding[0], it will only remove the last [0], so the elementDefinition.path will be Observation.identifer[0].type.coding and it should be Observation.identifier.type.coding ... I'll fix it in this PR.

Reply:
Ok actually we need to set the `elementDefinition` `id` attribute the right way, because it will define the `name` attribute in the `ElementNode`

Maybe you could use the split method with a regex like so `[[]\d+[]]` and then join them with an empty string